File: examples/upstream.js
import * as http from "http";
import Express from "express";

const app = Express();
app.get("/", (req, res) => {
    console.log(req.headers);
    const header = req.headers['x-client-tls-info'];
    if (header) {
        const info = JSON.parse(Buffer.from(header, 'base64').toString('utf8'));
        res.send(`Client Subject: ${info.subject}`);
    } else {
        res.status(401).send('Unauthorized');
    }
})

const server = http.createServer(app);
server.listen(8080);

File: examples/upstream.py
import base64
import json
from flask import Flask, request

app = Flask(__name__)

@app.route('/')
def index():
    header = request.headers.get('X-Client-TLS-Info', '')
    if header:
        info = json.loads(base64.b64decode(header).decode('utf-8'))
        client_subject = info.get('subject', None)
        return f"Client Subject: {client_subject}"
    return "Unauthorized", 401

app.run(port=8080, debug=True)

